includes/discovery/functions.inc.php

```diff
@@ -951,6 +951,27 @@ function discovery_process(&$valid, $device, $sensor_class, $pre_cache)
                 d_echo("Final sensor value: $value\n");
 
                 $skippedFromYaml = YamlDiscovery::canSkipItem($value, $index, $data, $sensor_options, $pre_cache);
+
+                // Check if we have a "num_oid" value. If not, we'll try to compute it from textual OIDs with snmptranslate.
+                if (empty($data['num_oid'])) {
```

Comment:
Can the duplicated code be avoided?

Reply:
It can, but we have to decide how : 
We should completely get rid of the code outside the YamlDiscovery (because discovery_process() and YamlDiscovery class are redundant for a big part).
Should I directly go that route (which means the title of this PR must change). Or do we get this PR as it is, and I start a new one to call YamlDiscovery class from discovery_process() and remove all possible redundant code there ?

Reply:
@PipoCanaja if you are willing to do the second PR, that would be greatly appreciated.  I'm fine with merging this as-is.

Reply:
OK. Let's do that then.